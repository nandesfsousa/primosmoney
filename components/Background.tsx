import * as React from 'react';
import { ImageBackground, StyleSheet, KeyboardAvoidingView } from 'react-native'

export const Background: React.FC = (props) =>{
    return (
        <ImageBackground
            source={require('../assets/images/background.png')}
            resizeMode="cover"
            style={styles.background}
        >
            <KeyboardAvoidingView style={styles.container} behavior="padding">
                {props.children}
            </KeyboardAvoidingView>
        </ImageBackground>
    );
}

const styles = StyleSheet.create({
    background: {
        flex: 1,
        width: '100%',
    },
    container: {
        flex: 1,
        padding: 20,
        width: '100%',
    },
})