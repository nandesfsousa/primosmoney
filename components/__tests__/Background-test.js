import * as React from 'react';
import renderer from 'react-test-renderer';

import { MonoText } from '../StyledText';
import { Background } from '../Background';

it(`renders correctly`, () => {
  const tree = renderer.create(<Background><MonoText>Snapshot test!</MonoText></Background>).toJSON();

  expect(tree).toMatchSnapshot();
});
