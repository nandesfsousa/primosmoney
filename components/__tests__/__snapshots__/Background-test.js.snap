// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#ffe002",
          },
          Array [
            undefined,
            Object {
              "fontFamily": "space-mono",
            },
          ],
        ]
      }
    >
      Snapshot test!
    </Text>
  </View>
</View>
`;
